packages/babel-helper-module-transforms/src/index.js

```diff
@@ -153,75 +145,51 @@ export function buildNamespaceInitStatements(
   return statements;
 }
 
-const moduleHeader = template(`
-  Object.defineProperty(EXPORTS, "__esModule", {
-    value: true,
-  })
-`);
-
-const moduleHeaderLoose = template(`
-  EXPORTS.__esModule = true;
-`);
-
 /**
  * Build an "__esModule" header statement setting the property on a given object.
  */
 function buildESModuleHeader(
   metadata: ModuleMetadata,
   enumerable: boolean = false,
 ) {
-  if (enumerable) {
-    return moduleHeaderLoose({
-      EXPORTS: t.identifier(metadata.exportName),
-    });
-  }
-
-  return moduleHeader({
-    EXPORTS: t.identifier(metadata.exportName),
-  });
+  return (enumerable
+    ? template.statement`
+        EXPORTS.__esModule = true;
+      `
+    : template.statement`
+        Object.defineProperty(EXPORTS, "__esModule", {
+          value: true,
+        });
+      `)({ EXPORTS: metadata.exportName });
 }
 
-const namespaceReexport = template(`
-  Object.keys(NAMESPACE).forEach(function(key) {
-    if (key === "default" || key === "__esModule") return;
-    VERIFY_NAME_LIST;
-
-    Object.defineProperty(EXPORTS, key, {
-      enumerable: true,
-      get: function() {
-        return NAMESPACE[key];
-      },
-    });
-  });
-`);
-const buildNameListCheck = template(`
-  if (Object.prototype.hasOwnProperty.call(EXPORTS_LIST, key)) return;
-`);
-
 /**
  * Create a re-export initialization loop for a specific imported namespace.
  */
 function buildNamespaceReexport(metadata, namespace) {
-  return namespaceReexport({
-    NAMESPACE: t.identifier(namespace),
-    EXPORTS: t.identifier(metadata.exportName),
+  return template.statement`
+    Object.keys(NAMESPACE).forEach(function(key) {
+      if (key === "default" || key === "__esModule") return;
+      VERIFY_NAME_LIST;
+
+      Object.defineProperty(EXPORTS, key, {
+        enumerable: true,
+        get: function() {
+          return NAMESPACE[key];
+        },
+      });
+    });
+  `({
+    NAMESPACE: namespace,
+    EXPORTS: metadata.exportName,
     VERIFY_NAME_LIST: metadata.exportNameListName
-      ? buildNameListCheck({
-          EXPORTS_LIST: t.identifier(metadata.exportNameListName),
-        })
+      ? template`
```

Comment:
is this also a `template.statement` or it doesn't matter?

Reply:
The default is essentially autodetect `statement` or `statements` so I really didn't need to change the other cases either. Guess I just didn't do this one.